compiler/rustc_pattern_analysis/src/constructor.rs

```diff
@@ -866,13 +874,27 @@ impl ConstructorSet {
         }
 
         match self {
-            ConstructorSet::Single { empty } => {
+            ConstructorSet::Struct { empty } => {
                 if !seen.is_empty() {
-                    present.push(Single);
+                    present.push(Struct);
                 } else if *empty {
-                    missing_empty.push(Single);
+                    missing_empty.push(Struct);
+                } else {
+                    missing.push(Struct);
+                }
+            }
+            ConstructorSet::Ref => {
+                if !seen.is_empty() {
+                    present.push(Ref);
+                } else {
+                    missing.push(Ref);
+                }
+            }
+            ConstructorSet::Union => {
```

Comment:
Can unions be uninhabited? 🤔

Reply:
Not according to `inhabited_predicate`. There's a comment "For now, unions are always considered inhabited" there which I'm inclined to trust

Reply:
Validity for unions is pretty undecided territory